spec/models/journal/aggregated_journal_spec.rb

```diff
@@ -0,0 +1,202 @@
+
+      expect(work_package.update_by!(new_author, changes)).to be_truthy
+    end
+
+    context 'by author of last change' do
+      let(:new_author) { initial_author }
+
+      it 'returns a single aggregated journal' do
+        expect(subject.count).to eql 1
+        expect(subject.first).to be_equivalent_to_journal work_package.journals.second
+      end
+
+      it 'is the initial journal' do
+        expect(subject.first.initial?).to be_truthy
+      end
+
+      context 'with a comment' do
+        let(:notes) { 'This is why I changed it.' }
+
+        it 'returns a single aggregated journal' do
+          expect(subject.count).to eql 1
+          expect(subject.first).to be_equivalent_to_journal work_package.journals.second
+        end
+
+        context 'adding a second comment' do
+          before do
+            expect(work_package.update_by!(new_author, notes: notes)).to be_truthy
+          end
+
+          it 'returns two journals' do
+            expect(subject.count).to eql 2
+            expect(subject.first).to be_equivalent_to_journal work_package.journals.second
+            expect(subject.second).to be_equivalent_to_journal work_package.journals.last
+          end
+
+          it 'has one initial journal and one non-initial journal' do
+            expect(subject.first.initial?).to be_truthy
+            expect(subject.second.initial?).to be_falsey
+          end
+        end
+
+        context 'adding another change without comment' do
+          before do
+            work_package.reload # need to update the lock_version, avoiding StaleObjectError
+            expect(work_package.update_by!(new_author, subject: 'foo')).to be_truthy
+          end
+
+          it 'returns a single journal' do
+            expect(subject.count).to eql 1
+          end
+
+          it 'combines the notes of the earlier journal with attributes of the later journal' do
+            expected_journal = work_package.journals.last
+            expected_journal.notes = work_package.journals.second.notes
+
+            expect(subject.first).to be_equivalent_to_journal expected_journal
+          end
+
+          it 'indicates the ID of the earlier journal via notes_id' do
+            expect(subject.first.notes_id).to eql work_package.journals.second.id
+          end
+        end
+      end
+    end
+
+    context 'by a different author' do
+      let(:new_author) { user2 }
+
+      it 'returns both journals' do
+        expect(subject.count).to eql 2
+        expect(subject.first).to be_equivalent_to_journal work_package.journals.first
+        expect(subject.second).to be_equivalent_to_journal work_package.journals.second
+      end
+    end
+  end
+
+  context 'WP updated after aggregation timeout expired' do
+    before do
+      work_package.status = FactoryGirl.build(:status)
+      work_package.save!
+      work_package.journals.second.created_at += 1.day # one day delay should always be long enough
```

Comment:
Why don't you use something like ``Setting.journal_aggregation_time_minutes + 1``?